--- origsrc/jack-audio-connection-kit-0.109.2/jackd/clientengine.c	2008-01-30 12:23:51.000000000 -0600
+++ src/jack-audio-connection-kit-0.109.2/jackd/clientengine.c	2008-09-22 13:46:45.944622800 -0500
@@ -229,7 +229,11 @@
 	const char *errstr;
 	char path_to_so[PATH_MAX+1];
 
+#ifdef __CYGWIN__
+	snprintf (path_to_so, sizeof (path_to_so), ADDON_DIR "/%s.dll", so_name);
+#else
 	snprintf (path_to_so, sizeof (path_to_so), ADDON_DIR "/%s.so", so_name);
+#endif
 	client->handle = dlopen (path_to_so, RTLD_NOW|RTLD_GLOBAL);
 	
 	if (client->handle == 0) {
--- origsrc/jack-audio-connection-kit-0.109.2/jackd/jackd.c	2008-01-30 12:23:51.000000000 -0600
+++ src/jack-audio-connection-kit-0.109.2/jackd/jackd.c	2008-09-22 14:10:58.716007800 -0500
@@ -325,7 +325,11 @@
 			continue;
 		}
 		ptr++;
+#ifdef __CYGWIN__
+		if (strncmp ("dll", ptr, 3) != 0) {
+#else
 		if (strncmp ("so", ptr, 2) != 0) {
+#endif
 			continue;
 		}
 
